server2/Arcadia.Assistant/Arcadia.Assistant.Notifications/Models/NotificationSettings.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace Arcadia.Assistant.Notifications.Models
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Fabric.Description;
+    using System.Linq;
+    using System.Runtime.Serialization.Json;
+    using System.Text;
+    using System.Xml;
+    using System.Xml.Linq;
+
+    using Contracts.Models;
+
+    using Interfaces;
+
+    public class NotificationSettings : INotificationSettings
+    {
+        public NotificationSettings(ConfigurationSection configurationSection)
+        {
+            if (bool.TryParse(configurationSection.Parameters["EnablePush"].Value, out var enable))
+            {
+                this.EnablePush = enable;
+            }
+
+            var map = configurationSection.Parameters["ClientNotificationProviders"].Value;
+            if (!string.IsNullOrEmpty(map))
+            {
+                var parsedMap = this.LoadNotificationProvidersMap(map);
+                this.ClientNotificationProvidersMap = parsedMap
+                    .GroupBy(x => x.Item1)
+                    .ToDictionary(x => x.Key,
+                        x => x.SelectMany(v => v.Item2).Distinct());
+            }
+        }
+
+        public bool EnablePush { get; }
+
+        public Dictionary<string, IEnumerable<NotificationType>> ClientNotificationProvidersMap { get; } = new Dictionary<string, IEnumerable<NotificationType>>();
```

Comment:
-> `IEnumerable<(string, IEnumerable<NotificationType>)>`. And in method body too.

Reply:
I prefere to use IDictionary in this case.

Reply:
Fixed.